### DISCUSSION TITLE
Themes don't use tailwinds colors configuration?

### DISCUSSION THREAD

**Discussion by USER-5** [skeletonlabs/skeleton#336] [Ideas]
While using skeleton, I noticed that including a theme would add the colour definitions as a raw CSS stylesheet. This meant that tailwind language server wasn't able to process the new colours, and didn't give me hints for them.

Is there a particular reason that this library defines colours in a CSS stylesheet, rather than using [tailwind's theming configuration](https://tailwindcss.com/docs/theme) settings?

**Comment by endigo9740** [comment]
Hey @USER-5, just a couple things to note here. First off, we do exactly what you suggest, just through our plugin:

https://github.com/Brain-Bones/skeleton/blob/dev/src/lib/tailwind/theme.cjs

The theme stylesheet supplies the color values as RGB, which then extends the Tailwind config theme colors. This is the process Tailwind recommends here:

https://tailwindcss.com/docs/customizing-colors#using-css-variables

As for why hints are not given, I'm not sure off hand. I'm so used to typing the values it hadn't even occurred to me it wasn't working! I could explore that a bit. I did note they are inserting the CSS custom properties in the `@base` directive in the link above, which we are not doing. Perhaps that's the issue?

I will be revisiting themes next week as part of this issue, so I'll aim to review this while I'm at it:
https://github.com/Brain-Bones/skeleton/issues/301

**Reply by USER-5** [reply]
Sounds good!

There is another reason for raising this - I'm trying to compile my app to a webcomponent which thus far has been difficult at every stage. But I think the :root{} selector isn't playing nice with the shadow Dom. I wouldn't put this at a high priority for most users though.

**Reply by endigo9740** [reply]
@USER-5 Out of curiosity I just tested this in the project and auto-suggestion is working fine for me. Are you sure you have your 
extension installed and setup properly? Your project configured properly?

![Screen Shot 2022-10-07 at 10 31 50 AM](https://user-images.githubusercontent.com/1509726/194592045-5f669a6e-d590-4ba3-b8d9-17b2fbe377b8.png)

Basically everything you asked for in the original message above is already done. I might suggest you give our documentation instructions a review once more and ensure you haven't missed a step.

Let me know if I can help debug.

**Reply by endigo9740** [reply]
If it helps, here's a complete Tailwind config in a minimal Skeleton app:

![Screen Shot 2022-10-07 at 10 38 34 AM](https://user-images.githubusercontent.com/1509726/194593420-2978ba40-5279-473c-a31e-ff92b72a7da6.png)

Here's the extension version I'm running in VS Code (v0.8.7)

![Screen Shot 2022-10-07 at 10 39 00 AM](https://user-images.githubusercontent.com/1509726/194593422-7abc039c-bbe7-4a8f-a029-a212fd0b3c30.png)

**Reply by USER-5** [reply]
Ok, I'm now fairly sure it has to do with the fact that I'm trying to compile it as a webcomponent. I've linked a couple of stackoverflow posts which might explain it better than I can. The change from :root to :host worked for me - but obviously I don't want to modify files within the node_modules folder.

https://stackoverflow.com/questions/42627939/css-selector-for-shadow-root-or-all-top-level-elements-in-shadow-root
https://stackoverflow.com/questions/48380267/css-variables-root-vs-host

**Reply by endigo9740** [reply]
Sorry, but unfortunately whatever you're doing doesn't seem to align with our goals for the project. We support SvelteKit, Vite, and Astro. For SvelteKit that means either the standard or SPA adapter. Anything beyond that is outside the scope of the project as it stands now. If you wish to compile a list of changes and make a feature request beyond that, then I'd encourage you to do so. But currently this doesn't sound like something we are keen to support at this time.

Good luck though. Keep in mind the project can be forked if you wish to modify things as a fundamental level.
